plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHivePartitionsOnDataLake.java

```diff
@@ -160,6 +194,83 @@ public void testInsertOverwritePartitionedAndBucketedExternalTable()
         assertOverwritePartition(externalTableName);
     }
 
+    @Test
+    public void testPartitionCacheDataFlush()
+    {
+        String tableName = "nation_" + randomTableSuffix();
+        String fullyQualifiedTestTableName = getTestTableName(tableName);
+
+        String partitionColumn = "regionkey";
+        String partitionValue = "0";
+
+        // Create table with partition on regionkey
+        computeActual(getCreateTableStatement(
+                fullyQualifiedTestTableName,
+                format("partitioned_by=ARRAY['%s']", partitionColumn)));
+        copyTpchNationToTable(fullyQualifiedTestTableName);
+
+        String queryUsingPartitionCache = format("SELECT name FROM %s WHERE %s=%s", fullyQualifiedTestTableName, partitionColumn, partitionValue);
+        String expectedQueryResult = "VALUES 'ALGERIA', 'MOROCCO', 'MOZAMBIQUE', 'ETHIOPIA', 'KENYA'";
+
+        // Fill partition cache and check we got expected results
+        assertQuery(queryUsingPartitionCache, expectedQueryResult);
+
+        // Copy partition to new location
+        renamePartitionResourcesOutsideTrino(tableName, partitionColumn, partitionValue);
+
+        // Should return 0 rows as we moved partition and cache is outdated. We use nonexistent partition
+        assertQueryReturnsEmptyResult(queryUsingPartitionCache);
+
+        // Should return expected rows when we refresh cache
+        //schema_name => 'dummy_schema', table_name => 'dummy_table', partition_column =>
+        getQueryRunner().execute(format(
```

Comment:
Here we are not testing that only cache for single partition was dropped. Can you populate cache for another partition before dropping cache for `0`. And see if querying another partition still returns cached data?

Reply:
Good point. Will fill cache for 2 partitions and clear 1 then verify only one query will succeed after flush.